mlir/lib/Dialect/Vector/Transforms/VectorDropLeadUnitDim.cpp

```diff
@@ -197,6 +197,23 @@ struct CastAwayInsertLeadingOneDim : public OpRewritePattern<vector::InsertOp> {
   }
 };
 
+static Value processTransferMask(OpBuilder &b, Location loc, Value mask,
+                                 VectorType newType, AffineMap newMap,
+                                 VectorType oldMaskType) {
+  // Infer the type of the new mask from the new map.
+  auto newMaskType = inferTransferOpMaskType(newType, newMap);
```

Comment:
`inferTransferOpMaskType` was introduced for verification purposes of the xfer ops' masks but I think it's ok to expose it as a utility.

Reply:
nit: spell out `auto`